File: src/main/resources/output/IFC4X3_js/Entities/IfcOwnerHistory/IfcOwnerHistory.js
class IfcOwnerHistory {
    constructor() {
        /** @type {IFCPERSONANDORGANIZATION} */
        this.OwningUser = null;
        /** @type {IFCAPPLICATION} */
        this.OwningApplication = null;
        /** @type {IFCSTATEENUM} */
        this.State = null;
        /** @type {IFCCHANGEACTIONENUM} */
        this.ChangeAction = null;
        /** @type {IFCTIMESTAMP} */
        this.LastModifiedDate = null;
        /** @type {IFCPERSONANDORGANIZATION} */
        this.LastModifyingUser = null;
        /** @type {IFCAPPLICATION} */
        this.LastModifyingApplication = null;
        /** @type {IFCTIMESTAMP} */
        this.CreationDate = null;
    }

    // === WHERE CLAUSES ===
    // CorrectChangeAction : (EXISTS(LastModifiedDate)) OR (NOT(EXISTS(LastModifiedDate)) AND NOT(EXISTS(ChangeAction))) OR (NOT(EXISTS(LastModifiedDate)) AND EXISTS(ChangeAction) AND ((ChangeAction = IfcChangeActionEnum.NOTDEFINED) OR (ChangeAction = IfcChangeActionEnum.NOCHANGE)))
}

File: src/main/resources/output/IFC4X3_vbnet/Entities/IfcOwnerHistory/IfcOwnerHistory.vb
Public Class IfcOwnerHistory
    Public OwningUser As IfcPersonAndOrganization
    Public OwningApplication As IfcApplication
    Public State As IfcStateEnum
    Public ChangeAction As IfcChangeActionEnum
    Public LastModifiedDate As IfcTimeStamp
    Public LastModifyingUser As IfcPersonAndOrganization
    Public LastModifyingApplication As IfcApplication
    Public CreationDate As IfcTimeStamp

    ' === WHERE CLAUSES ===
    ' CorrectChangeAction : (EXISTS(LastModifiedDate)) OR (NOT(EXISTS(LastModifiedDate)) AND NOT(EXISTS(ChangeAction))) OR (NOT(EXISTS(LastModifiedDate)) AND EXISTS(ChangeAction) AND ((ChangeAction = IfcChangeActionEnum.NOTDEFINED) OR (ChangeAction = IfcChangeActionEnum.NOCHANGE)))
End Class
